p2p/src/peer_manager/global_ip.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2021 Protocol Labs.
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Based on libp2p sources:
+// https://github.com/libp2p/rust-libp2p/blob/73cbbe29679f5d56d81d5477d3796e82712f9ac2/protocols/autonat/src/behaviour.rs
+
+pub trait GlobalIp {
```

Comment:
I think this trait should be called `IsGlobalIp`. This is because in its current form, it sounds like it's assuming that a type that implements it is a global ip, but it just adds the ability to check that property.